.github/workflows/release.yml

```diff
@@ -0,0 +1,53 @@
+name: Release
+
+on:
+  release:
+    types: [ created ]
+
+permissions:
+  contents: write
+  packages: write
+
+jobs:
+  publish_backend:
+    runs-on: "ubuntu-latest"
+    defaults:
+      run:
+        working-directory: backend
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build and publish backend
+        uses: macbre/push-to-ghcr@master
+        with:
+          context: ./backend/
+          dockerfile: ./backend/Dockerfile
+          image_name: incompetent-hosting-provider/ihp-backend
+          github_token: ${{ secrets.GH_API_TOKEN_GHPR }}
+
+  publish_cli:
+    runs-on: "ubuntu-latest"
+    strategy:
+      matrix:
+        os: [linux, windows, macos]
```

Comment:
Use two matrix values instead of the same job twice.
See here
https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy

Reply:
Applied multi dimension matrix.